consensus/service/src/consensus_service.rs

```diff
@@ -606,18 +606,26 @@ impl<
                         .map(|x| format!("{:X?}", x.id.0))
                         .map_err(|e| log::error!(logger, "Error getting block {} {:?}", b - 1, e))
                         .ok();
-                    latest_block_timestamp = ledger_db
-                        .get_block_signature(b - 1)
-                        .map(|x| x.signed_at())
-                        .map_err(|e| {
+
+                    latest_block_timestamp = match ledger_db.get_block_signature(b - 1) {
+                        Ok(x) => Some(x.signed_at()),
+                        // Note, a node may not write a block signature for the case where it is not
```

Comment:
// Note, a block signature will not be present if it was not processed by an enclave
 // participating in consensus. For example, unsigned blocks can be created by
 // a validator node that falls behind its peers and enters into catchup.

Reply:
Love it - thank you!